Bottom: d70b0ad7ac594e2078272201e9a8554f1a994e19
Top:    2831a0ffd17d656eea1ec9fab8b9f3ca48e6d9e2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-10 17:04:17 +0000

Revert "Fix CenterWindow, DRC activation pointer warping."

This reverts commit f12d98e90025544995d87a02de3a79d78a536e6c.

CenterWindow didn't do what it's name said, it just warped the pointer.
Now it does both for the gtk hid, so DRC checking works consistently
when the violation is off-screen or not.  It's still broken in lesstif
HID but no worse than it was, and there's a warning about the
pan-and-warp action not being implemented there.


---

diff --git a/src/hid.h b/src/hid.h
index b9e9b80..57d047f 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -521,13 +521,9 @@ typedef enum
        * adjusted so that the cursor and the crosshair are at the same
        * point on the screen.
        */
-#define HID_SC_DO_NOTHING                          0
-
-#define HID_SC_WARP_POINTER	                       1
-
-#define HID_SC_PAN_VIEWPORT                        2
-
-#define HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER 3
+#define HID_SC_DO_NOTHING	0
+#define HID_SC_WARP_POINTER	1
+#define HID_SC_PAN_VIEWPORT	2
 
       hidval (*add_timer) (void (*func) (hidval user_data_),
 			   unsigned long milliseconds_, hidval user_data_);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index a96c41a..d49421b 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -39,7 +39,6 @@ pan_common (GHidPort *port)
   ghidgui->adjustment_changed_holdoff = FALSE;
 
   ghid_port_ranges_changed();
-  gdk_window_process_all_updates ();
 }
 
 static void
@@ -304,8 +303,7 @@ ghid_set_crosshair (int x, int y, int action)
     }
 
   if (action != HID_SC_PAN_VIEWPORT &&
-      action != HID_SC_WARP_POINTER &&
-      action != HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER)
+      action != HID_SC_WARP_POINTER)
     return;
 
   /* Find out where the drawing area is on the screen. gdk_display_get_pointer
@@ -314,27 +312,9 @@ ghid_set_crosshair (int x, int y, int action)
    */
   gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
                          &offset_x, &offset_y);
-
   display = gdk_display_get_default ();
-  screen = gdk_display_get_default_screen (display);
 
   switch (action) {
-
-    case HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER:
-
-      // Center the viewport on the crosshair
-      ghid_pan_view_abs (gport->crosshair_x - gport->view.width / 2,
-                         gport->crosshair_y - gport->view.height / 2,
-                         0, 0);
-
-      // Warp pointer to crosshair
-      ghid_pcb_to_event_coords (gport->crosshair_x, gport->crosshair_y,
-                                &widget_x, &widget_y);
-      gdk_display_warp_pointer (display, screen, widget_x + offset_x,
-                                widget_y + offset_y);
-
-      break;
-
     case HID_SC_PAN_VIEWPORT:
       /* Pan the board in the viewport so that the crosshair (who's location
        * relative on the board was set above) lands where the pointer is.
diff --git a/src/hid/gtk/gui-drc-window.c b/src/hid/gtk/gui-drc-window.c
index 5d1c014..8f5a352 100644
--- a/src/hid/gtk/gui-drc-window.c
+++ b/src/hid/gtk/gui-drc-window.c
@@ -168,7 +168,6 @@ row_activated_cb (GtkTreeView *view, GtkTreePath *path,
     return;
 
   CenterDisplay (violation->x_coord, violation->y_coord);
-  gtk_window_present (GTK_WINDOW (gport->top_window));
 }
 
 
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 2cb3393..a04ab62 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -3535,16 +3535,6 @@ lesstif_set_crosshair (int x, int y, int action)
 
     }
 
-  if (action == HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER)
-    {
-      fprintf (
-          stderr,
-          "warning:%s:%i: HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER not "
-          "implemented in this HID, using HID_SC_WARP_POINTER instead\n",
-          __FILE__,
-          __LINE__ );
-      action = HID_SC_WARP_POINTER;
-    }
   if (action == HID_SC_PAN_VIEWPORT)
     {
       Window root, child;
diff --git a/src/misc.c b/src/misc.c
index 87693ae..bf2cce2 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -834,15 +834,10 @@ void
 CenterDisplay (Coord X, Coord Y)
 {
   Coord save_grid = PCB->Grid;
-
   PCB->Grid = 1;
-
   if (MoveCrosshairAbsolute (X, Y))
     notify_crosshair_change (true);
-
-  gui->set_crosshair (Crosshair.X, Crosshair.Y,
-                      HID_SC_CENTER_IN_VIEWPORT_AND_WARP_POINTER);
-
+  gui->set_crosshair (Crosshair.X, Crosshair.Y, HID_SC_WARP_POINTER);
   PCB->Grid = save_grid;
 }
